mavros_extras/src/plugins/debug.cpp

```diff
@@ -0,0 +1,254 @@
+/**
+ * @brief Debug messages plugin
+ * @file debug.cpp
+ * @author Nuno Marques <n.marques21@hotmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2017 Nuno Marques.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+
+#include <mavros_msgs/Debug.h>
+
+namespace mavros {
+namespace extra_plugins {
+//! @brief Plugin for Debug msgs from MAVLink API
+class DebugPlugin : public plugin::PluginBase {
+public:
+	DebugPlugin() : PluginBase(),
+		debug_nh("~debug"),
+		publish_debug(false)
+	{ }
+
+	void initialize(UAS &uas_)
+	{
+		PluginBase::initialize(uas_);
+
+		// subscribers
+		debug_sub = debug_nh.subscribe("send", 10, &DebugPlugin::debug_cb, this);
+
+		debug_nh.param("publish_debug", publish_debug, false);
```

Comment:
That param are useless. I don't think that anyone need that. Just unblacklist and viola.

Reply:
I explain you my idea: if `publish_debug` is false, the plugin still runs but if one wants to check the outputs, one needs to activate the DEBUG output (`rxconsole` or something). If it is true, it publishes to the topics. I'm going to add `ROS_DEBUG_STREAM_COND_NAMED` (checks if `publish_debug` is false, and if yes it publishes the output to DEBUG console).